src/Sources/Sources.jl

```diff
@@ -73,6 +74,88 @@ function get_int_coalescence(mom_p::Array{FT}, ODE_parameters::Dict, ker::Kernel
   return coal_int
 end
 
+"""
+get_int_coalescence_two_modes(mom_p::Array{Real}, ODE_parameters::Dict, ker::KernelTensor{Real})
+
+  - `mom_p` - prognostic moments of the two particle mass distributions
+  - `ODE_parameters` - ODE parameters, a Dict containing a key ":dist" which 
+                       contains the two distributions at the previous time step. dists is an 
+                       array of ParticleDistribution; it is used to calculate the 
+                       diagnostic moments and also to dispatch to the
+                       moments-to-parameters mapping (done by the function
+                       moments_to_params) for the given types of distribution
+  - `ker` - coalescence kernel tensor
+Returns the coalescence integral for all moments in `mom_p`.
+"""
+function get_int_coalescence_two_modes(mom_p::Array{FT}, ODE_parameters::Dict, ker::KernelTensor{FT}) where {FT <: Real}
+  r = ker.r
+
+  dist_prev = ODE_parameters[:dist]
+  n_params = [nparams(dist_prev[i]) for i in 1:2]
+  mom_p_ = [mom_p[1:n_params[1]], mom_p[n_params[1]+1:end]]
+  s = [length(mom_p_[i]) for i in 1:2]
+
+  # Need to build diagnostic moments
+  dist = [moments_to_params(dist_prev[i], mom_p_[i]) for i in 1:2]
```

Comment:
Ideally we do not allocate, but simply update throughout this entire function, including here

Reply:
Leave for future PR; see issue #86